hw_8/hw_8.2.js

```diff
@@ -0,0 +1,284 @@
+	const idArr = [];
+	companyArr.forEach(comp => {
+		idArr.push(comp.id);
+		comp.departments.forEach(depart => idArr.push(depart.id));
+	});
+	const maxId = Math.max(...idArr);
+	return maxId;
+}
+
+
+const addNewCompany = companyName => {
+	const newComp = {
+		id: getMaxId(enterprises) + 1,
+		name: companyName,
+		departments: [],
+	}
+	enterprises.push(newComp);
+}
+
+addNewCompany('Название нового предприятия');
+console.log(enterprises);
+
+
+/*
+4. Написать функцию, которая будет добавлять отдел в предприятие. В качестве аргумента принимает id предприятия, в которое будет добавлен отдел и название отдела.
+Пример:
+addDepartment(1, "Название нового отдела")
+*/
+
+function getCompanyByCompanyId(compId) {
+	let compObj; 
+	enterprises.find(comp => comp.id === compId ? compObj = comp : false);
+	return compObj;
+}
+
+
+const addNewDepartment = (compId, departName) => {
+	const newDepart = {
+		id: getMaxId(enterprises) + 1,
+		name: departName,
+		employees_count: 0,
+	}
+	getCompanyByCompanyId(compId).departments.push(newDepart);
+}
+
+addNewDepartment(11, 'Название нового отдела');
+addNewDepartment(11, 'Название super отдела');
+console.log(enterprises[3]);
+
+
+/*
+5. Написать функцию для редактирования названия предприятия. Принимает в качестве аргумента id предприятия и новое имя предприятия.
+Пример:
+editEnterprise(1, "Новое название предприятия")
+*/
+
+const editCompanyName = (compId, newCompName) => {
+	getCompanyByCompanyId(compId).name = newCompName;
+}
+
+editCompanyName(11, 'Предприятие 4');
+console.log(enterprises[3]);
+
+
+/*
+6. Написать функцию для редактирования названия отдела. Принимает в качестве аргумента id отдела и новое имя отдела.
+Пример:
+editDepartment(7, "Новое название отдела")
+*/
+
+function getDepartByDepartId(departId) {
+	let departObj; 
+	enterprises.find(comp => comp.departments.find(depart => depart.id === departId ? departObj = depart : false));
+	return departObj;
+}
+
+
+const editDepartName = (departId, newDepartName) => {
+	getDepartByDepartId(departId).name = newDepartName;
+}
+
+editDepartName(12, 'Финансовый отдел');
+editDepartName(13, 'Коммерческий отдел');
+console.log(enterprises[3]);
+
+
+/*
+7. Написать функцию для удаления предприятия. В качестве аргумента принимает id предприятия.
+Пример:
+deleteEnterprise(1)
+*/
+
+const deleteCompanyItem = compId => {
+	const compIndex = enterprises.findIndex(comp => comp.id === compId);
+	compIndex != -1 ? enterprises.splice(compIndex, 1) : false;
+}
+
+deleteCompanyItem(5);
+console.log(enterprises);
+
+
+/*
+8. Написать функцию для удаления отдела. В качестве аргумента принимает id отдела. Удалить отдел можно только, если в нем нет сотрудников.
+Пример:
+deleteDepartment(3)
+*/
+
+const deleteDepartItem = departId => {
+	enterprises.forEach(comp => {
+		const departIndex = comp.departments.findIndex(depart => depart.id === departId && depart.employees_count == 0);
+		departIndex != -1 ? comp.departments.splice(departIndex, 1) : false;
```

Comment:
```suggestion
		if(departIndex != -1) comp.departments.splice(departIndex, 1)
```
Better not to use ternary operator without any real "else" case

Reply:
Or maybe add throw error `Department with id '${departId}' was not found`
Same for previous tasks